Fix issues uncovered by xfstests

rfjakob opened the pull request:
Adds the `FsName` member to `MountOptions` and uses it in the loopback example

Before:
    
    Filesystem                 Type                   1K-blocks      Used  Available Use% Mounted on
    pathfs.pathInode           fuse.pathfs.pathInode    4015252   3219628     795624  81% /tmp/mountpoint
    
After:
    
    Filesystem                 Type           1K-blocks      Used  Available Use% Mounted on
    /tmp/original              fuse.loopback    4015252   3219620     795632  81% /tmp/mountpoint

Also, I enabled ClientInodes. This is in preparation of getting xfstests to run on loopbackfs.

rfjakob commented:
I signed it!

On Sat, Sep 19, 2015 at 12:48 PM, googlebot <notifications@github.com>
wrote:

> Thanks for your pull request. It looks like this may be your first
> contribution to a Google open source project, in which case you'll need to
> sign a Contributor License Agreement (CLA).
>
> [image: :memo:] *Please visit https://cla.developers.google.com/
> <https://cla.developers.google.com/> to sign.*
>
> Once you've signed, please reply here (e.g. I signed it!) and we'll
> verify. Thanks.
> ------------------------------
>
>    - If you've already signed a CLA, it's possible we don't have your
>    GitHub username or you're using a different email address. Check your
>    existing CLA data <https://cla.developers.google.com/clas> and verify
>    that your email is set on your git commits
>    <https://help.github.com/articles/setting-your-email-in-git/>.
>    - If you signed the CLA as a corporation, please let us know the
>    company's name.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hanwen/go-fuse/pull/53#issuecomment-141644128>.
>

hanwen commented:
thanks for all these changes. I will try to look soon, but am rather busy.

I think we probably don't want to set ClientInodes as a default, because it has subtle semantics. LoopbackFS uses path names as identifier, which is fundamentally at odds with posix if you look at the details (files with hardlinks and such.)

rfjakob commented:
Hi there, I have addressed your comments and rebased to master.
I have also dropped the ClientInodes patch for now. I will add a flag to enable it at run-time next week.

rfjakob commented:
Already merged, closing the pull request.
